pipeline/transformer/transformer_calm/src/main/scala/uk/ac/wellcome/platform/transformer/calm/CalmTransformer.scala

```diff
@@ -38,51 +44,66 @@ object CalmTransformer
     "third-party metadata"
   )
 
-  override def apply(record: CalmRecord, version: Int): Result[Work[Source]] =
-    if (shouldSuppress(record)) {
-      Right(
-        Work.Deleted[Source](
-          state = Source(sourceIdentifier(record), record.retrievedAt),
-          data = workData(record).getOrElse(WorkData[DataState.Unidentified]()),
-          version = version,
-          deletedReason = Some(SuppressedFromSource("Calm"))
-        )
-      )
-    } else {
-      workData(record) match {
-        case Right(data) =>
-          Right(
-            Work.Visible[Source](
-              state = Source(sourceIdentifier(record), record.retrievedAt),
-              version = version,
-              data = data
-            ))
+  override def apply(sourceData: CalmSourceData,
+                     version: Int): Result[Work[Source]] = sourceData match {
+    case CalmSourceData(record, true) =>
+      Right(deletedWork(version, record, DeletedFromSource("Calm")))
+    case CalmSourceData(record, false) if shouldSuppress(record) =>
+      Right(deletedWork(version, record, SuppressedFromSource("Calm")))
+    case CalmSourceData(record, false) =>
+      tryParseValidWork(record, version)
```

Comment:
I don't love the dangling booleans here, they're never great for readability IMO.

```suggestion
    case CalmSourceData(record, isDeleted) if isDeleted =>
      Right(deletedWork(version, record, DeletedFromSource("Calm")))
    case CalmSourceData(record, _) if shouldSuppress(record) =>
      Right(deletedWork(version, record, SuppressedFromSource("Calm")))
    case CalmSourceData(record, _) =>
      tryParseValidWork(record, version)
```